details/detail.js

```diff
@@ -0,0 +1,14 @@
+import { getSpotsByID } from '../fetch-utils.js';
+import { renderSpotKind } from '../render-utils.js';
+
+const params = new URLSearchParams(window.location.search);
+const spotDetail = document.getElementById('spot-detail');
+
+async function loadData() {
+    const data = await getSpotsByID(params.get('id'));
```

Comment:
It can sometimes be difficult to read code where a function call (`params.get('id')`) is passed as an argument to another function call (`getSpotsByID()`).

Consider separating this out into two lines to improve readability. Something like this:

```js
const id = params.get('id');
const data = await getSpotsByID(id);
```